rental.rb

```diff
@@ -0,0 +1,18 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
+
+  def belongs_to?
+    @book = self
+  end
+
+  def belongs_to_a_person?
+    @person = self
+  end
```

Comment:
- [ ] Please check below :point_down: my suggestion to correct your code

```suggestion
  attr_accessor :date, :person, :book

  def initialize(date, person, book)
    @date = date
    @book = book
    book.rental.push(self) unless book.rental.include?(self)
    @person = person
    person.rental.push(self) unless person.rental.include?(self)
  end
```

- **attr_accessor :date, :person, :book :**
  - This line creates getter and setter methods for the instance variables date, person, and book.

- **def initialize(date, person, book):**
  - This is the constructor (initializer) method for the Rental class. It takes three arguments: date, person, and book.

- **@date = date:**
  - This assigns the value of the date argument to the instance variable @date.

- **@book = book:**
  - This assigns the value of the book argument to the instance variable @book.

- **book.rentals.push(self) unless book.rentals.include?(self):**
  - This line adds the current Rental instance (self) to the rentals array of the associated book (book).
  - The unless book.rentals.include?(self) part ensures that the rental is not added multiple times to the book.rentals array.

- **@person = person:**
  - This assigns the value of the person argument to the instance variable @person.

- **person.rentals.push(self) unless person.rentals.include?(self):**
  - This line adds the current Rental instance (self) to the rentals array of the associated person (person).
  - Similar to the previous case, unless person.rentals.include?(self) prevents duplicate entries in the person.rentals array.

In summary, this initialize method creates a Rental instance and associates it with both a person and a book. It ensures that the rental is added to the corresponding arrays of rentals for the associated person and book, avoiding duplicates. This allows you to track which rentals are associated with each person and book.